### PR TITLE
Fix GAP.Packages.install with a HACK

### LINKED ISSUE #452: Travis: compiling `profiling` package fails due to missing autoconf
The following has been happening on Travis for some time:
```
Running 'make' 'clean' 
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/travis/.julia/artifacts/848f0f318405266896d24d11e4baf5cf7bf3f203/gap-4.11.0/pkg/profiling-2.2.1/cnf/missing aclocal-1.16 -I m4
/home/travis/.julia/artifacts/848f0f318405266896d24d11e4baf5cf7bf3f203/gap-4.11.0/pkg/profiling-2.2.1/cnf/missing: line 81: aclocal-1.16: command not found
WARNING: 'aclocal-1.16' is missing on your system.
You should only need it if you modified 'acinclude.m4' or
'configure.ac' or m4 files included by 'configure.ac'.
The 'aclocal' program is part of the GNU Automake package:
<https://www.gnu.org/software/automake>
It also requires GNU Autoconf, GNU m4 and Perl in order to run:
<https://www.gnu.org/software/autoconf>
<https://www.gnu.org/software/m4/>
<https://www.perl.org/>
Makefile:416: recipe for target 'aclocal.m4' failed
make: *** [aclocal.m4] Error 127
```
Of course we could work around this by e.g. installing autoconf. But there are two underlying issues I'd rather fix:

1. Why don't the Travis job fail??? I mean, it's "nice" that I can ignore this issue for now and focus on more important things, but in general I'd prefer to notice these things.

2. To fix it, we could modify the GAP tarball once again... But I'd prefer to instead switch to using the PackageManager to install these (see PR #413 and also #365)

### PR DESCRIPTION
**Pull request by fingolfin**
Fixes #452
Closes #467

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -66,6 +66,13 @@ end
 # clean out some clutter
 rm(joinpath(gap_bin_root, "obj"), force=true, recursive=true)
 
+# copy bin/BuildPackages.sh to work around PackageManager issues
+# with out-of-tree builds.
+# FIXME: remove this one PackageManager is improved
```
Comment (mohamed-barakat) [inline deps/build.jl:71]:
```suggestion
# FIXME: remove this once PackageManager is improved
```

**Comment by mohamed-barakat** [comment]
The following failed:

```julia
julia> GAP.Packages.install("io")
#I  Getting PackageInfo URLs...
#I  Created directory /Users/mo/.julia/dev/GAP/deps/build/gap-4.11.0-julia-1.4.0/pkg/
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/io/PackageInfo.g ...
#I  Downloading archive from URL https://github.com/gap-packages/io/releases/download/v4.7.0/io-4.7.0.tar.gz ...
#I  Saved archive to /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmJRWQuN/io-4.7.0.tar.gz
#I  Extracting to /Users/mo/.julia/dev/GAP/deps/build/gap-4.11.0-julia-1.4.0/pkg/io-4.7.0 ...
#I  Running compilation script on /Users/mo/.julia/dev/GAP/deps/build/gap-4.11.0-julia-1.4.0/pkg/io-4.7.0 ...
#I  Possible error detected: see log at /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmpS3y8c/exec-log.txt
#I  Compilation failed (package may still be usable)
#I  Package availability test failed
#I  (for IO 4.7.0)
#I  Removed directory /Users/mo/.julia/dev/GAP/deps/build/gap-4.11.0-julia-1.4.0/pkg/io-4.7.0
false
```

```
$ cat /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmpS3y8c/exec-log.txt
/bin/sh: /Users/mo/.julia/dev/GAP/deps/build/gap-4.11.0-julia-1.4.0/bin/BuildPackages.sh: No such file or directory
```

**Comment by mohamed-barakat** [comment]
Thank you for supporting the optional argument `pkgdir` :)

**Comment by fingolfin** [comment]
@mohamed-barakat tests for this finally pass. Perhaps you can to try it again?

**Comment by mohamed-barakat** [comment]
Compiling `io` succeeded but compiling `dgraphs` failed.

```julia
julia> GAP.Packages.install("io")
#I  Getting PackageInfo URLs...
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/io/PackageInfo.g ...
#I  Downloading archive from URL https://github.com/gap-packages/io/releases/download/v4.7.0/io-4.7.0.tar.gz ...
#I  Saved archive to /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmuuTeUm/io-4.7.0.tar.gz
#I  Extracting to /Users/mo/.gap/pkg/io-4.7.0 ...
true

julia> GAP.Packages.install("io")
#I  Getting PackageInfo URLs...
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/io/PackageInfo.g ...
#I  The newest version of package "io" is already installed
true

julia> GAP.Packages.install("digraphs")
#I  Getting PackageInfo URLs...
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/Digraphs/PackageInfo.g ...
#I  Downloading archive from URL https://github.com/gap-packages/Digraphs/releases/download/v1.2.0/digraphs-1.2.0.tar.gz ...
#I  Saved archive to /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmZNitPP/digraphs-1.2.0.tar.gz
#I  Extracting to /Users/mo/.gap/pkg/digraphs-1.2.0 ...
#I  Checking dependencies for Digraphs...
#I    io >=4.5.1: true
#I    orb >=4.8.2: true
#I    datastructures >=0.2.5: false
#I  Retrieving PackageInfo.g from https://gap-packages.github.io/datastructures/PackageInfo.g ...
#I  Installing dependency datastructures 0.2.5 ...
#I  Downloading archive from URL https://github.com/gap-packages/datastructures/releases/download/v0.2.5/datastructures-0.2.5.tar.gz ...
#I  Saved archive to /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmBbHLyj/datastructures-0.2.5.tar.gz
#I  Extracting to /Users/mo/.gap/pkg/datastructures-0.2.5 ...
#I  Checking dependencies for datastructures...
#I    GAPDoc 1.5: true
#I  Running compilation script on /Users/mo/.gap/pkg/datastructures-0.2.5 ...
#I  Possible error detected: see log at /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmZMOECA/exec-log.txt
#I  Compilation failed (package may still be usable)
#I  Checking dependencies for datastructures...
#I    GAPDoc 1.5: true
#I  Package availability test failed
#I  (for datastructures 0.2.5)
#I  Removed directory /Users/mo/.gap/pkg/datastructures-0.2.5
#I  Dependencies not satisfied for digraphs-1.2.0
#I  Removed directory /Users/mo/.gap/pkg/digraphs-1.2.0
false
```

**Comment by fingolfin** [comment]
So what does the log at /var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmZMOECA/exec-log.txt say?

**Comment by mohamed-barakat** [comment]
Dump of `/var/folders/z7/br6dwhjj5xn8691ygcnbzxc40000gs/T//tmZMOECA/exec-log.txt`:

```
Using GAP root /Users/mo/.julia/artifacts/848f0f318405266896d24d11e4baf5cf7bf3f203/gap-4.11.0
ERROR: /Users/mo/.julia/artifacts/848f0f318405266896d24d11e4baf5cf7bf3f203/gap-4.11.0 is not the root of a gap installation (no sysinfo.gap)
ERROR: Please provide the absolute path of your GAP root directory as
ERROR: first argument with '--with-gaproot=' to this script.
```

**Comment by fingolfin** [comment]
That's odd: the given path is into the artifact and indeed not the correct GAPROOT. But the main point of this PR is to solve this kind of problem, by (a) using a newer version of `BuildPackages.sh`.

I tried replicating this, but failed. In particular, it works fine on my Mac, on plesken and in a ubuntu docker container (which I just had running for your issue #465 anyway).

@mohamed-barakat Could you please provide some details about the system where the issue appears?

**Comment by fingolfin** [comment]
@mohamed-barakat I am going to merge this anyway, as it seems to improve lots of things and doesn't make anything worse. If you can still reproduce the digraphs problems afterwards, please open an issue for it (with some details on the system where it occurs), then I can work on fixing it in a new PR.

**Comment by mohamed-barakat** [comment]
I was probably doing something wrong. It is now working.

**Comment by mohamed-barakat** [comment]
Now that `GAP.Packages.install` is working I would like to start using it in `HomalgProject.jl`. Is it possible to make a new minor release?

**Comment by fingolfin** [comment]
@mohamed-barakat: done, GAP.jl 0.4.1 is tagged and should be available soon

**Comment by mohamed-barakat** [comment]
Wonderful, thanks :)

**Pull request merged**